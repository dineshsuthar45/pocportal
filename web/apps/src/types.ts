import { InputProps, TypographyProps } from "@mui/material";
import { CSSProperties, ChangeEvent } from "react";

export interface IThemeSchema {
  palette: any;
  typography: any;
  components?: any;
}

export interface IGlobalTableProps {
  row: any;
  list: any;
}

export interface IGlobalButtonProps {
  label: any;
  sublabel?: string;
  disabled?: boolean;
  onClick?: () => void;
  style?: CSSProperties;
  type?: string;
}

export interface IGlobalImgProps {
  img: string;
  alt: string;
  style?: CSSProperties;
}

export interface IGlobaltextFieldProps {
  id?: string;
  value: string;
  label: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  name: string;
  errmsg?: string;
  type?: string;
  disabled?: boolean;
  style?: CSSProperties;
  placeholder?: string;
  showPassword?: boolean;
  InputProps?: InputProps;
}

export interface IGlobalTypoProps {
  id?: number;
  variant: TypographyProps["variant"];
  label: string;
  sublabel?: string;
  style?: CSSProperties;
}
