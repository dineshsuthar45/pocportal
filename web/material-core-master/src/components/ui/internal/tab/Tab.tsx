import { styled, useTheme } from "@mui/material";
import MuiTab from "@mui/material/Tab";
import classNames from "classnames";
import "./Tab.scss";
import { TabProps } from "./Tab.types";

const StyledTab = styled(MuiTab)<TabProps>((prop) => ({
  "&:hover:not(.Mui-selected)": {
    color: prop.theme.palette[prop.color ?? "primary"].main,
  },
}));

/**
 * Further properties can be found in MUI website at https://mui.com/material-ui/api/tab/
 */
export const MTab = ({
  color = "primary",
  disabled = false,
  disableFocusRipple = false,
  disableRipple = false,
  iconPosition = "top",
  wrapped = false,
  ...props
}: TabProps) => {
  const theme = useTheme();

  return (
    <StyledTab
      className={classNames("tab-root", theme.palette.mode)}
      color={color}
      disabled={disabled}
      disableFocusRipple={disableFocusRipple}
      disableRipple={disableRipple}
      iconPosition={iconPosition}
      wrapped={wrapped}
      {...props}
    />
  );
};

MTab.defaultProps = {
  color: "primary",
  disabled: false,
  disableFocusRipple: false,
  disableRipple: false,
  iconPosition: "top",
  wrapped: false,
};
