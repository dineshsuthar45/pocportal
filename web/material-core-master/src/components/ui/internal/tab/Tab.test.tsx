import { render } from "@testing-library/react";
import React from "react";
import { MTab } from "./Tab";
import { TabProps } from "./Tab.types";

describe("Tab", () => {
  const renderComponent = () => render(<MTab {...props} />);
  let props: TabProps;

  beforeEach(() => {
    props = {
      disabled: false,
      disableFocusRipple: false,
      disableRipple: false,
      value: 1,
      wrapped: false,
      label: "Tab label",
    };
  });

  it("should render correctly", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
