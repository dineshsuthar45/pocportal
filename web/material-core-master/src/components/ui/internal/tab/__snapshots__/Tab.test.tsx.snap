// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab should render correctly 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit tab-root light css-xj3am1-MuiButtonBase-root-MuiTab-root"
    color="primary"
    role="tab"
    tabindex="-1"
    type="button"
  >
    Tab label
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
